### PR TITLE
PTFM-20076: Add dx-toolkit support for downloadRestricted flag

### PR DESCRIPTION
**Pull request by robosoto**
Add --download-restricted flag to dx-toolkit.

Test plan: run_integration_tests.sh --client_tests_only --tests test_dxpy.TestDXProject

### CODE REVIEW

File: src/python/dxpy/bindings/dxproject.py [hunk 1]
```diff
@@ -296,6 +296,8 @@ def new(self, name, summary=None, description=None, protected=None,
         :type protected: boolean
         :param restricted: If provided, whether the project should be restricted
         :type restricted: boolean
+        :param download_restricted: If provided, whether the project should be download restricted
```
Comment (nicolasbockg) [inline src/python/dxpy/bindings/dxproject.py]: Consider rewording this as: "If provided, whether external downloads should be restricted"

File: src/python/dxpy/bindings/dxproject.py [hunk 2]
```diff
@@ -348,6 +352,8 @@ def update(self, name=None, summary=None, description=None, protected=None,
...
```
Comment (nicolasbockg) [inline src/python/dxpy/bindings/dxproject.py]: Same here.

File: src/python/dxpy/bindings/dxproject.py (hunk 1 above)
Comment (robosoto) [inline src/python/dxpy/bindings/dxproject.py]: Good call

**Comment by nicolasbockg** [comment]
LGTM. 
@geetduggal Just FYI, we are adding a `downloadRestricted` Flag to projects.

**Pull request merged**